docs/source/triggers.rst

```diff
@@ -3,15 +3,62 @@ Triggers
+| | :class:`~cocotb.triggers.RisingEdge`   | | Resume after a *rising* edge on a single-bit signal                                     |
+| | :class:`~cocotb.triggers.FallingEdge`  | | Resume after a *falling* edge on a single-bit signal                                    |
+| | :class:`~cocotb.triggers.ValueChange`  | | Resume after *any* edge on a signal                                                     |
+| | :class:`~cocotb.triggers.ClockCycles`  | | Resume after the specified number of transitions of a signal                            |
+| | :class:`~cocotb.triggers.Edge`         | | (deprecated, see :class:`~cocotb.triggers.ValueChange`)                                 |
++------------------------------------------+-------------------------------------------------------------------------------------------+
+| **Timing Triggers**                                                                                                                  |
++------------------------------------------+-------------------------------------------------------------------------------------------+
+| | :class:`~cocotb.triggers.Timer`        | | Resume after the specified time                                                         |
+| | :class:`~cocotb.triggers.ReadOnly`     | | Resume when the simulation timestep moves to the *read-only* phase                      |
+| | :class:`~cocotb.triggers.ReadWrite`    | | Resume when the simulation timestep moves to the *read-write* phase                     |
+| | :class:`~cocotb.triggers.NextTimeStep` | | Resume when the simulation timestep moves to the *next time step*                       |
++------------------------------------------+-------------------------------------------------------------------------------------------+
+| **Concurrency Triggers**                                                                                                             |
++------------------------------------------+-------------------------------------------------------------------------------------------+
+| | :func:`~cocotb.triggers.gather`        | | Resume after *all* given Tasks or Triggers                                              |
+| | :func:`~cocotb.triggers.select`        | | Resume after *any* given Tasks or Triggers                                              |
+| | :func:`~cocotb.triggers.wait`          | | Await on all given Tasks or Triggers concurrently and block until a condition is met    |
+| | :class:`~cocotb.triggers.NullTrigger`  | | Resume immediately                                                                      |
+| | :class:`~cocotb.triggers.Combine`      | | (deprecated, see :func:`~cocotb.triggers.gather`)                                       |
+| | :class:`~cocotb.triggers.First`        | | (deprecated, see :func:`~cocotb.triggers.select`)                                       |
```

Comment:
These are not necessarily deprecated. They still have value is producing things that look and act like triggers that are a combination of other triggers. They are deprecated only for passing Tasks.